### PR TITLE
React-Redux integration.

### PR DESCRIPTION
**Pull request by tzvaita**
In this milestone, we: 
- Created all the requested directories.
- Implemented App, Booklist, Book, and BooksForm components.
- Defined Redux actions and reducers.
- Initialized the Redux store.

Thanks for reviewing.

Somoye and Tennyson

### CODE REVIEW

**Review by mikenath223 — CHANGES_REQUESTED** [review]
### Hi Team 👋

**Good work on this.👍**

- [ ] Kindly note comments on your code.

Kindly fix these issues and request another review👍
Feel free to contact me on slack(michgolden) if you need further clarification.

HAPPY CODING👨‍💻

**Comment by mikenath223** [comment]
- Code still in review.

File: src/common/constantModule.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+const constant = () => {
+  const categories = ['Action', 'Biography', 'History', 'Horror', 'Kids', 'Learning', 'Sci-Fi'];
+  const defaultState = [
+    {
+      id: Math.floor(Math.random() * 1000),
+      title: 'Somoye and his projects',
+      category: 'Action',
+    },
+    {
+      id: Math.floor(Math.random() * 1000),
```
Comment (mikenath223) [inline src/common/constantModule.js:10]: - [ ] Consider wrapping the randomId generator as a function and calling it in each case.

File: src/containers/BooksForm.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+import Constants from '../common/constantModule';
+
+const BooksForm = () => (
+  <form className="bookform-section">
+    <h4>ADD NEW BOOK</h4>
+    <input type="text" />
```
Comment (mikenath223) [inline src/containers/BooksForm.js:7]: - [ ] The BookForm should contain the handleChange function and call it on the form input.

File: src/containers/BooksForm.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+import Constants from '../common/constantModule';
+
+const BooksForm = () => (
+  <form className="bookform-section">
```
Comment (mikenath223) [inline src/containers/BooksForm.js:5]: - [ ] The BookForm should contain the handleSubmit function and should be attached to the form elemnt.

File: src/containers/BooksList.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import Book from '../components/Book';
+
+const BooksList = ({ books }) => {
+  const booksList = books.map(book => <Book book={book} key={Math.random()} />);
```
Comment (mikenath223) [inline src/containers/BooksList.js:7]: - [ ] The BookList should contain the handle remove function and attached to each Book component.

File: src/common/constantModule.js [hunk 5]
```diff
@@ -0,0 +1,21 @@
+const constant = () => {
+  const categories = ['Action', 'Biography', 'History', 'Horror', 'Kids', 'Learning', 'Sci-Fi'];
+  const defaultState = [
+    {
+      id: Math.floor(Math.random() * 1000),
```
Comment (mikenath223) [inline src/common/constantModule.js:5]: - [ ] - OPTIONAL -  Kindly wrap this randomId generator in a function and call it on each id.

**Review by whiz25 — APPROVED** [review]
## Approved :clap: 
All seems okay. Merge and proceed to the next milestone.

### Martin Afani

**Pull request merged**